### PR TITLE
Change inherent overlap error to a warning for now, to ease the breakage

### LINKED ISSUE #32247: 'duplicate definitions' regression with winapi 0.2.5
With `rustc 1.9.0-nightly (74b886ab1 2016-03-13)` I see this error building winapi 0.2.5:

```
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:148:13: 150:14 error: duplicate definitions with name `item`: [E0201]
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:148             pub unsafe fn $variant(&self) -> &$fieldtype {
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:149                 ::std::mem::transmute(&self.$field)
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:150             }
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\commctrl.rs:2496:1: 2496:59 note: in this expansion of UNION! (defined in C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs)
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:148:13: 150:14 help: run `rustc --explain E0201` to see a detailed explanation
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:148:13: 150:14 note: conflicting definition is here:
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:148             pub unsafe fn $variant(&self) -> &$fieldtype {
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:149                 ::std::mem::transmute(&self.$field)
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:150             }
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\commctrl.rs:2503:1: 2503:59 note: in this expansion of UNION! (defined in C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs)
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:152:13: 154:14 error: duplicate definitions with name `item_mut`: [E0201]
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:152             pub unsafe fn $variantmut(&mut self) -> &mut $fieldtype {
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:153                 ::std::mem::transmute(&mut self.$field)
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:154             }
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\commctrl.rs:2496:1: 2496:59 note: in this expansion of UNION! (defined in C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs)
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:152:13: 154:14 help: run `rustc --explain E0201` to see a detailed explanation
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:152:13: 154:14 note: conflicting definition is here:
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:152             pub unsafe fn $variantmut(&mut self) -> &mut $fieldtype {
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:153                 ::std::mem::transmute(&mut self.$field)
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs:154             }
C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\commctrl.rs:2503:1: 2503:59 note: in this expansion of UNION! (defined in C:\Users\brian\AppData\Local\.multirust\cargo\registry\src\github.com-88ac128001ac3a9a\winapi-0.2.5\src\macros.rs)
error: aborting due to 2 previous errors

```

This worked as recently as `rustc 1.9.0-nightly (c9629d61c 2016-03-10)`.

cc @retep998 @nikomatsakis

### PR DESCRIPTION
**Pull request by aturon**
Closes #32247

r? @nikomatsakis

### CODE REVIEW

**Comment by nikomatsakis** [comment]
So we usually use "forward compat" lints for these warnings now. It should be easy to modify, just grep around for `MATCH_OF_UNIT_VARIANT_VIA_PAREN_DOTDOT` and follow its lead. There are a few things you have to do:

```
// define the lint and register it:
librustc/lint/builtin.rs:140:9:    pub MATCH_OF_UNIT_VARIANT_VIA_PAREN_DOTDOT,
librustc/lint/builtin.rs:184:13:            MATCH_OF_UNIT_VARIANT_VIA_PAREN_DOTDOT,

// define the forward compat details
librustc_lint/lib.rs:170:28:            id: LintId::of(MATCH_OF_UNIT_VARIANT_VIA_PAREN_DOTDOT),

// issue the lint
librustc_typeck/check/_match.rs:605:38:        sess.add_lint(lint::builtin::MATCH_OF_UNIT_VARIANT_VIA_PAREN_DOTDOT,
```

**Comment by aturon** [comment]
Done; re-r?

File: src/librustc_typeck/coherence/overlap.rs [hunk 1]
```diff
@@ -68,11 +69,15 @@ impl<'cx, 'tcx> OverlapChecker<'cx, 'tcx> {
...
-                    err.emit();
+                    let msg = format!("duplicate definitions with name `{}`", name);
+                    let node_id = self.tcx.map.as_local_node_id(item1.def_id()).unwrap();
+                    self.tcx.sess.add_lint(lint::builtin::OVERLAPPING_INHERENT_IMPLS,
+                                           node_id,
+                                           self.tcx.span_of_impl(item1.def_id()).unwrap(),
+                                           msg);
+                    // span_note!(&mut warn, self.span_of_def_id(item2.def_id()),
+                    //            "conflicting definition is here:");
+                    // warn.emit();
```
Comment (nikomatsakis) [inline src/librustc_typeck/coherence/overlap.rs]: ah well, probably obvious most of the time?

**Comment by aturon** [comment]
Oops, errant commented-out code included.

High priority due to wanting to hit nightly ASAP.

@bors: r=nikomatsakis p=1

**Comment by eddyb** [comment]
I may have broken @bors. Trying to work around that:
@bors r- clear

**Pull request merged**